app/models/finance/statement_line_item.rb

```diff
@@ -13,16 +13,23 @@ class Finance::StatementLineItem < ApplicationRecord
     paid: "paid",
     voided: "voided",
     ineligible: "ineligible",
+    awaiting_clawback: "awaiting_clawback",
+    clawed_back: "clawed_back",
   }
 
   scope :billable, -> { where(state: %w[eligible payable paid]) }
+  scope :refundable, -> { where(state: %w[awaiting_clawback clawed_back]) }
 
   validate :validate_single_billable_relationship, on: [:create]
 
+  def billable?
+    %w[eligible payable paid].include?(state)
+  end
+
 private
 
   def validate_single_billable_relationship
-    if Finance::StatementLineItem
+    if billable? && Finance::StatementLineItem
```

Comment:
What do you need to check for `Finance::StatementLineItem`?

Reply:
i want to make sure a declaration does not get associated to more that one statement for each direction ie:
- if money going out (eligible, payable, paid) we at most only have one line item
- if money coming in (awaiting_clawback, clawed_back) we at most only have one line item

Reply:
Ah, sorry, I misread that. I didn't see it was a multi-line condition.